manager/assets/modext/widgets/core/tree/modx.tree.js

```diff
@@ -145,13 +145,50 @@ Ext.extend(MODx.tree.Tree,Ext.tree.TreePanel,{
     ,options: {}
     ,disableHref: false
 
-    ,onLoad: function(ldr,node,resp) {
+    ,onLoad: function (ldr,node,resp) {
         // no select() here, just addClass, using Active Input Cookie Value to set focus
-        Ext.each(node.childNodes, function(node){
+        Ext.each(node.childNodes,function (node) {
             if (node.attributes.selected) {
                 //node.select();
                 node.ui.addClass('x-tree-selected');
             }
+
+            var elId = node.ui.elNode.id + '_tools';
+            var el = document.createElement('div');
+            el.id = elId;
+            el.className = 'modx-tree-node-btn-create';
+
+            // for now we only add a create icon to documents
+            if (node.attributes.classKey == "modDocument") {
```

Comment:
I've not loaded the PR locally yet, but it looks like this may be preventing it from adding the icon on weblinks or custom resource types (collections, moregallery, etc)? Perhaps this onLoad method should call a new method on the tree like `this.prepareChildNode(node)` which the resource tree object can extend to add this button?

Reply:
Well, that was on purpose. I did not see a realistic use case where e.g. one would add a child to a weblink. Chris already mentioned Collections to be a "problematic" case. I will see what I can do here with exts-fu - probably coming back to you for help with that.

Reply:
On some sites I've seen a parent weblink pointing to the first child (although there's other ways to go about that too), so I don't think it should be making too many judgements on where it can/cannot work.

Reply:
Yeah, doing this right now. Due to performance reasons I also thought about adding the buttons only on hover of the tree nodes (and not in advance for all nodes), but I don't now currently (thanks to the wonderful easily to understand extjs) where to start that.